File: public/dsa/searching/non_repeating_element/hashing/non_repeating_element.cpp

#include <bits/stdc++.h>
using namespace std;

int findNonRepeatingElement(vector<int> &arr) {
    unordered_map<int, int> mp;
    for (int v : arr) {
        mp[v]++;
    }
    for (auto [k, v] : mp) {
        if (v == 1) {
            return k;
        }
    }
    return -1;
}

// tests-start
int main() {
    vector<int> arr = {1, 1, 2, 3, 3, 4, 4};
    cout << findNonRepeatingElement(arr) << endl;
    return 0;
}
// tests-end

/*output
2
*/

File: public/dsa/searching/non_repeating_element/hashing/non_repeating_element.py
from collections import defaultdict
from typing import List


class Solution:
    def findNonRepeatingElement(self, arr: List[int]) -> int:
        mp = defaultdict(int)
        for v in arr:
            mp[v] += 1
        for k in mp:
            if mp[k] == 1:
                return k


# tests-start
arr = [1, 1, 2, 3, 3, 4, 4]
sol = Solution()
print(sol.findNonRepeatingElement(arr))
# tests-end

'''output
2
'''
